Modules/core/Controller/CoretilesController.php

```diff
@@ -99,4 +107,99 @@ public function docAction(){
+        $modelSpacePending = new CorePendingAccount();
+        $data = $modelSpacePending->getSpaceIdsForPending($_SESSION["id_user"]);
+        $userPendingSpaceIds = array();
+
+        if ($data && count($data) > 0) {
+            foreach ($data as $space) {
+                array_push($userPendingSpaceIds, $space["id_space"]);
+            }
+        }
+
+        $modelSpaceUser = new CoreSpaceUser();
+        $data = $modelSpaceUser->getUserSpaceInfo($_SESSION["id_user"]);
+        $userSpaceIds = array();
+        $spacesUserIsAdminOf = array();
+
+        if ($data && count($data) > 0) {
+            foreach ($data as $space) {
+                array_push($userSpaceIds, $space["id_space"]);
+                if ($space["status"] === "4") {
+                    array_push($spacesUserIsAdminOf, $space["id_space"]);
+                }
+            }
+        }
+        $result = array(
+            "userSpaceIds" => $userSpaceIds,
+            "userPendingSpaceIds" => $userPendingSpaceIds,
+            "spacesUserIsAdminOf" => $spacesUserIsAdminOf
+        );
+
+        return $result;
+    }
+
+    /**
+     * 
+     * Manage actions resulting from user request to join or leave a space
+     *
+     * @param int $id_space
+     * @param bool $isMemberOfSpace
+     */
+    public function selfJoinSpaceAction($id_space) {
+        $modelSpaceUser = new CoreSpaceUser();
+        $isMemberOfSpace = $modelSpaceUser->exists($_SESSION["id_user"], $id_space);
+        if ($isMemberOfSpace) {
+            $modelSpaceUser = new CoreSpaceUser();
+            $modelSpaceUser->delete($_SESSION["id_user"], $id_space);
+        } else if (!$isMemberOfSpace) {
+            $spaceModel = new CoreSpace();
+            $spaceName = $spaceModel->getSpaceName($id_space);
+            $modelSpacePending = new CorePendingAccount();
+            $modelSpacePending->add($_SESSION["id_user"], $id_space);
+            $mailParams = [
+                "id_space" => $id_space,
+                "space_name" => $spaceName
+            ];
+            $this->NotifyAdminsByEmail($mailParams, "new_join_request");
+        } 
+        $this->redirect("coretiles");
+    }
+
+    /**
+     * 
+     * Send an Email to space managers (status > 2) notifying that logged user requested to join space
+     *
+     * @param array $params required to fill sendEmail() parameters. Depends on why we want to notify space admins
+     * @param string $origin determines how to get sendEmail() paramters from $params
+     */
+    public function NotifyAdminsByEmail($params, $origin) {
+        $lang = $this->getLanguage();
+        if ($origin === "new_join_request") {
+            $spaceModel = new CoreSpace();
+            $emailSpaceManagers = $spaceModel->getEmailsSpaceManagers($params["id_space"]);
+            $mailer = new MailerSend();
+            $from = Configuration::get('smtp_from');
+            $fromName = "Platform-Manager";
+            $spaceName = $params["space_name"];
```

Comment:
same remak on space_name, manage case where not in params

$spaceName = $params["space_name"] ?? "";